File: bindings/go/geometry.go
package maplibre

// LatLng is a geographic coordinate in degrees.
type LatLng struct {
	Latitude  float64
	Longitude float64
}

// ScreenPoint is a logical pixel coordinate.
type ScreenPoint struct {
	X float64
	Y float64
}

// EdgeInsets is a screen-space inset in logical map pixels.
type EdgeInsets struct {
	Top    float64
	Left   float64
	Bottom float64
	Right  float64
}

// ProjectedMeters is a spherical Mercator coordinate in meters.
type ProjectedMeters struct {
	Northing float64
	Easting  float64
}

// Vec3 is a three-component vector.
type Vec3 struct {
	X float64
	Y float64
	Z float64
}

// Quaternion stores x, y, z, w components.
type Quaternion struct {
	X float64
	Y float64
	Z float64
	W float64
}

// LatLngBounds is a geographic bounds rectangle in degrees.
type LatLngBounds struct {
	Southwest LatLng
	Northeast LatLng
}

// CanonicalTileID identifies one canonical tile.
type CanonicalTileID struct {
	Z uint32
	X uint32
	Y uint32
}

File: bindings/swift/Sources/MaplibreNative/Geometry.swift

public struct LatLng: Equatable, Sendable {
  public var latitude: Double
  public var longitude: Double

  public init(latitude: Double, longitude: Double) {
    self.latitude = latitude
    self.longitude = longitude
  }

  init(native: NativeLatLng) {
    latitude = native.latitude
    longitude = native.longitude
  }

  var nativeInput: NativeLatLng {
    NativeLatLng(latitude: latitude, longitude: longitude)
  }
}

public struct ScreenPoint: Equatable, Sendable {
  public var x: Double
  public var y: Double

  public init(x: Double, y: Double) {
    self.x = x
    self.y = y
  }

  init(native: NativeScreenPoint) {
    x = native.x
    y = native.y
  }

  var nativeInput: NativeScreenPoint {
    NativeScreenPoint(x: x, y: y)
  }
}

public struct LatLngBounds: Equatable, Sendable {
  public var southwest: LatLng
  public var northeast: LatLng

  public init(southwest: LatLng, northeast: LatLng) {
    self.southwest = southwest
    self.northeast = northeast
  }

  init(native: NativeLatLngBounds) {
    southwest = LatLng(native: native.southwest)
    northeast = LatLng(native: native.northeast)
  }

  var nativeInput: NativeLatLngBounds {
    NativeLatLngBounds(
      southwest: southwest.nativeInput,
      northeast: northeast.nativeInput
    )
  }
}

public struct EdgeInsets: Equatable, Sendable {
  public var top: Double
  public var left: Double
  public var bottom: Double
  public var right: Double

  public init(top: Double, left: Double, bottom: Double, right: Double) {
    self.top = top
    self.left = left
    self.bottom = bottom
    self.right = right
  }

  init(native: NativeEdgeInsets) {
    top = native.top
    left = native.left
    bottom = native.bottom
    right = native.right
  }

  var nativeInput: NativeEdgeInsets {
    NativeEdgeInsets(top: top, left: left, bottom: bottom, right: right)
  }
}
